developer-docs-site/docs/guides/transaction-management.md

```diff
@@ -0,0 +1,97 @@
+# Transaction Management
+
+This guide explains how to build a transaction management harness that can scale on the Aptos blockchain.
+
+## Building a Transaction Manager
+
+This section covers the major aspects of building a transaction management harness. This consists of the following core components:
+
+- A sequence number generator that allocates and manages available sequence numbers for a single account.
+- A transaction manager that receives payloads from an application or a user, sequence numbers from the sequence number generator, and has access to the account key to combine the three pieces together into a viable signed transaction. It then also takes the responsibility for pushing the transaction to the blockchain.
+- An on-chain worker, leader harness that let’s multiple accounts share the signer of a single shared account.
+
+Currently this framework assumes that the network builds no substantial queue, that is a transaction that is submitted executes and commits with little to no delay. In order to address high demand, this work needs to be extended with the following components:
+
+- Optimizing `base_gas_unit` price to ensure priority transactions can be committed to the blockchain.
+- Further handling of transaction processing rates to ensure that the expiration timer is properly set.
+
+### Implementations
+
+- Python
+    - [Sequence number manager](https://github.com/aptos-labs/aptos-core/pull/7987)
+    - [Transaction manager](https://github.com/aptos-labs/aptos-core/pull/7987)
+- [Worker-leader smart contract](https://github.com/aptos-labs/aptos-core/pull/7986)
+
+### Managing Sequence Numbers
+
+Each transaction requires a distinct sequence number that is sequential to previously submitted transactions. This can be provided by the following process:
+
+1. At startup, query the blockchain for the account’s current sequence number.
+2. Support up to 100 transactions in flight at the same time, that is 100 sequence numbers can be allocated without confirming that any have been committed.
+3. If there are 100 transactions in flight, determine the actual committed state by querying the network. This will update the current sequence number.
+4. If there are less than 100 transactions in flight, return to step 2.
+5. Otherwise sleep for .1 seconds and continue to re-evaluate the current on-chain sequence number.
+6. All transactions should have an expiration time. If the expiration time has passed, assume that there has been a failure and reset the sequence number. The trivial case is to only monitor for failures when the maximum number of transactions are in flight and to let other services manages this otherwise.
```

Comment:
if you have small throughput, step 3 will never be called, and sequence number rechecked for successful transactions, so all transactions will be treated as failed? or is the order in step 6 the opposite - you first check the sequence number, and then treat some as succeeded and some as expired based on onchain seq_num and onchain_timestamp?

Reply:
So I need to add some more logic here and there, but the gist of it is that the actual manager should provide the feedback or the user can call synchronize

Reply:
so I added a new component that checks the first transaction that was logged and times it out if appropriate